src/qutip_tensornetwork/core/data/tensor_train/tensor_train.py

```diff
@@ -234,6 +234,119 @@ def _to_tt_format(self):
 
         self._nodes = set(nodes)
 
+    def truncate(self, bond_dimension=None, max_truncation_err=None):
+        """Truncate in-place the bond dimension of the tensor train according
+        to ``bond_dimension``. This is done from left to right and once
+        finished the network is right normalized.
+
+        Notes
+        -----
+        The truncation method consists on performing a svd decomposition from
+        left to right and truncating each nodes svd by the given parameters.
+
+        Parameters
+        ----------
+        bond_dimension: List of int or int
+            List of integers that define, from left to right, the target bond
+            dimension. If a single integer is provided is understood as the
+            same bond dimension for all bonds. None is understood as
+            "infinite" bond dimension and hence no truncation is done.
+
+        max_truncation_err: List of float or float
+            Maximum truncation error for each individual node. If a single
+            value is provided instead of a list, it is the assumed that the
+            same value applies for every node.
+
+        Returns
+        -------
+        error: float
+            List of truncated singular values.
+
+        See also
+        --------
+        tensornetwork.split_node:
+            Function employed to split individual nodes. ``bond_dimension``
+            and ``max_truncation_err`` directly translate into
+            ``max_singular_values`` and ``max_truncation_err``.
+        """
+        if not isinstance(bond_dimension, list):
+            bond_dimension = [bond_dimension] * len(self.bond_edges)
+
+        if not isinstance(max_truncation_err, list):
+            max_truncation_err = [max_truncation_err] * len(self.bond_edges)
+
+        if len(self.train_nodes) == 1:
+            return []
+
+        total_error = []
+        for bond_edge, dim, err in zip(
+            self.bond_edges, bond_dimension, max_truncation_err
+        ):
+
+            node = bond_edge.node1
+            next_node = bond_edge.node2
+
+            # We begin by performing the following (svd) transformation:
+            #   |   |             |           |
+            # - * - * -   --->  - * - * - * - * -
+            #   |   |             |           |
+            # where the edge between the nodes (left fig) is the bond_edge.
+            # Note that the svd truncates the dimension of the new nodes.
+            left_edges = [edge for edge in node if edge is not bond_edge]
+            right_edges = [bond_edge]
+            lnode, s, rnode, error = tn.split_node_full_svd(
+                node,
+                left_edges,
+                right_edges,
+                max_singular_values=dim,
+                max_truncation_err=err,
+            )
+            total_error += error.tolist()
```

Comment:
I wonder how useful it is to return a flat list here? If we return a flat list, then one gets `[err1, err2, err3, ...]` but one has no idea which edge the singular values came from because one doesn't know how many were truncated each time. Perhaps it would be better to do `total_error.append(error.tolist())`? Or `total_error.append(error)` if the error object is useful to have (maybe it's a tensor node)?